src/sandstorm/web-session.capnp

```diff
@@ -225,9 +313,16 @@ interface WebSession @0xa50711a14d35a8ce extends(Grain.UiSession) {
       noContent :group {
         # Return successful, but with no content (status codes 204 and 205)
 
+        statusCode @17 :SuccessCode;
+
         shouldResetForm @15 :Bool;
         # If this is the response to a form submission, should the form be reset to empty?
         # Distinguishes between HTTP response 204 (False) and 205 (True)
+
+        dav @20 :Text;
+        # Supported webdav modes
+
+        etag @21 :Text;
```

Comment:
Hmm, I think we may want:

    struct ETag {
      value @0 :Text;  # does not include quotes
      weak @1 :Bool;
    }

Otherwise it's weird that this field has to be formatted in a specific way (in quotes, with a possible "W/" prefix). The Cap'n Proto interface is supposed to encode all structure.

I guess for the preconditions, we'll want them to be `List(ETag)` rather than `List(Text)`.